### PR TITLE
Fixing incorrect verified notice behavior in tipping banner

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/brave/brave-browser/issues/6182

Screen Shot 2019-09-26 at 4 13 00 PM

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Defined in issue

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/ui/webui/brave_tip_ui.cc [hunk 1]
```diff
@@ -62,6 +62,10 @@ class RewardsTipDOMHandler : public WebUIMessageHandler,
   void OnGetRecurringTips(
       std::unique_ptr<brave_rewards::ContentSiteList> list);
   void TweetTip(const base::ListValue *args);
+  void GetExternalWallet(const base::ListValue* args);
+  void OnExternalWallet(
+      int32_t result,
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_tip_ui.cc]: const

File: browser/ui/webui/brave_tip_ui.cc [hunk 2]
```diff
@@ -421,3 +430,40 @@ void RewardsTipDOMHandler::FetchBalance(const base::ListValue* args) {
           weak_factory_.GetWeakPtr()));
   }
 }
+
+void RewardsTipDOMHandler::GetExternalWallet(
+    const base::ListValue* args) {
+  if (!rewards_service_)
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_tip_ui.cc]: please add brackets

File: browser/ui/webui/brave_tip_ui.cc [hunk 3]
```diff
@@ -421,3 +430,40 @@ void RewardsTipDOMHandler::FetchBalance(const base::ListValue* args) {
...
+  rewards_service_->GetExternalWallet(type,
+     base::BindOnce(
+         &RewardsTipDOMHandler::OnExternalWallet,
+         weak_factory_.GetWeakPtr()));
+}
+
+void RewardsTipDOMHandler::OnExternalWallet(
+    int32_t result,
+    std::unique_ptr<brave_rewards::ExternalWallet> wallet) {
+  if (!web_ui()->CanCallJavascript())
```
Comment (NejcZdovc) [inline browser/ui/webui/brave_tip_ui.cc]: please add brackets

File: browser/ui/webui/brave_tip_ui.cc (hunk 1 above)
Comment (ryanml) [reply browser/ui/webui/brave_tip_ui.cc]: done

File: browser/ui/webui/brave_tip_ui.cc (hunk 3 above)
Comment (ryanml) [reply browser/ui/webui/brave_tip_ui.cc]: done

**Comment by kjozwiak** [comment]
@mandar-brave @NejcZdovc @ryanml is there a reason this needs to be uplifted into `0.71.x`? I understand it's a regression but we're trying to limit uplifts for issues that are either critical or greatly reduce functionality.

**Comment by mandar-brave** [comment]
@kjozwiak basis its the first step in the process and being upfront is good. Step-1 is user tapping the BAT logo. I would prefer to keep it for this release knowing it is a 0.69 specific change.

**Comment by kjozwiak** [comment]
@mandar-brave if we want this in `0.69.x`, we'll need another PR for `0.69.x` as the following were the only ones creates:

* Fixing incorrect verified notice behavior in tipping banner (0.71) #3570
* Fixing incorrect verified notice behavior in tipping banner (0.70) #3571

At this point, we're cutting it pretty close here and landing something basically last minute.

**Comment by NejcZdovc** [comment]
@kjozwiak yeah because of close I only asked @ryanml to do 71 and 70. @mandar-brave let us know what to do

**Comment by mandar-brave** [comment]
@kjozwiak i understand the constraints so you make the final call on risk (it is not a blocker at my end but definitely prefer to see it in this release). I am responding from a product perspective. considering the text is newly introduced specifically for 0.69, it would definitely leave inconsistencies from the previous experience (where the user does not have to click send-tip before text to suggest tip may not be sent).

**Comment by rebron** [comment]
Not a blocker and we're too late on 69.x for the fix. Fix will be in 70.x though.

**Pull request merged**